Tone/core/clock/TickSource.ts

```diff
@@ -41,6 +53,16 @@ export class TickSource<TypeName extends "bpm" | "hertz"> extends ToneWithContex
 	 */
 	private _tickOffset: Timeline<TickSourceOffsetEvent> = new Timeline();
 
+	/**
+	 * Memoized values of getTicksAtTime at events with state other than "started"
+	 */
+	private _ticksAtTime: Timeline<TickSourceTicksAtTimeEvent> = new Timeline<TickSourceTicksAtTimeEvent>();
```

Comment:
Thinking about that assumption which is now baked in that the events added to the timeline are always increasing, you can pass in an argument to Timeline which will assert that each time a value is added. 

```ts
new Timeline({
 increasing: true
})
```

Reply:
I decided not to do add this assertion because it seems risky and might break things. Instead I changed the memoization to work with out-of-order events by invalidating all memoized values after the modified event.